packages/jest-worker/src/types.ts

```diff
@@ -27,98 +23,105 @@ export type PARENT_MESSAGE_ERROR =
+  uid?: number;
+  gid?: number;
 };
 
 export interface WorkerPoolInterface {
-  getStderr(): Readable;
-  getStdout(): Readable;
+  getStderr(): NodeJS.ReadableStream;
+  getStdout(): NodeJS.ReadableStream;
   getWorkers(): Array<WorkerInterface>;
-  createWorker(WorkerOptions): WorkerInterface;
-  send(number, ChildMessage, Function, Function): void;
+  createWorker(options: WorkerOptions): WorkerInterface;
+  send(
+    workerId: number,
+    request: ChildMessage,
+    onStart: OnStart,
+    onEnd: OnEnd,
+  ): void;
   end(): void;
 }
 
 export interface WorkerInterface {
-  send(ChildMessage, Function, Function): void;
+  send(
+    request: ChildMessage,
+    onProcessStart: OnStart,
+    onProcessEnd: OnEnd,
+  ): void;
   getWorkerId(): number;
-  getStderr(): Readable;
-  getStdout(): Readable;
-  onExit(number): void;
-  onMessage(any): void;
+  getStderr(): NodeJS.ReadableStream;
+  getStdout(): NodeJS.ReadableStream;
+  onExit(exitCode: number): void;
+  onMessage(message: ParentMessage): void;
 }
 
 export type FarmOptions = {
-  computeWorkerKey?: (string, ...Array<any>) => ?string,
-  exposedMethods?: $ReadOnlyArray<string>,
-  forkOptions?: ForkOptions,
-  setupArgs?: Array<mixed>,
-  maxRetries?: number,
-  numWorkers?: number,
+  computeWorkerKey?: (method: string, ...args: Array<unknown>) => string | null;
+  exposedMethods?: ReadonlyArray<string>;
+  forkOptions?: ForkOptions;
+  setupArgs?: Array<unknown>;
+  maxRetries?: number;
+  numWorkers?: number;
   WorkerPool?: (
     workerPath: string,
     options?: WorkerPoolOptions,
-  ) => WorkerPoolInterface,
-  enableWorkerThreads?: boolean,
+  ) => WorkerPoolInterface;
+  enableWorkerThreads?: boolean;
 };
 
-export type WorkerPoolOptions = {|
-  setupArgs: Array<mixed>,
-  forkOptions: ForkOptions,
-  maxRetries: number,
-  numWorkers: number,
-  enableWorkerThreads: boolean,
-|};
-
-export type WorkerOptions = {|
-  forkOptions: ForkOptions,
-  setupArgs: Array<mixed>,
-  maxRetries: number,
-  workerId: number,
-  workerPath: string,
-|};
+export type WorkerPoolOptions = {
+  setupArgs: Array<unknown>;
+  forkOptions: ForkOptions;
+  maxRetries: number;
+  numWorkers: number;
+  enableWorkerThreads: boolean;
+};
+
+export type WorkerOptions = {
+  forkOptions: ForkOptions;
+  setupArgs: Array<unknown>;
+  maxRetries: number;
+  workerId: number;
+  workerPath: string;
+};
 
 // Messages passed from the parent to the children.
 
-export type MessagePort = {
-  ...typeof EventEmitter,
-  postMessage(any): void,
+export type MessagePort = typeof EventEmitter & {
+  postMessage(message: unknown): void;
 };
 
 export type MessageChannel = {
-  port1: MessagePort,
-  port2: MessagePort,
+  port1: MessagePort;
+  port2: MessagePort;
 };
 
 export type ChildMessageInitialize = [
   typeof CHILD_MESSAGE_INITIALIZE, // type
   boolean, // processed
   string, // file
-  ?Array<mixed>, // setupArgs
-  ?MessagePort, // MessagePort
+  Array<unknown> | undefined, // setupArgs
```

Comment:
Intentionally left out `null`? I see `ChildMessageInitialize` being assigned from `any` and usually we have `null`s in this package

Edit: I guess for an array leaving out the element makes more sense though. On a side note, wondering if TS differentiates between `[string | undefined]` and `[string] | never[]`. Would matter for inferring the parameter type of `forEach` 😄

Reply:
yea, mostly because it didn't complain. the less allowed values the better, IMO. Easy enough to add more allowed types at some point if there's value in it

Reply:
Tried out the side note thing in the TS playground btw, looks like it does differentiate 😎 not relevant for our use case though